PermissionDenied.java

```diff
@@ -0,0 +1,115 @@
+package mage.cards.p;
+
+import java.util.UUID;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.abilities.Ability;
+import mage.abilities.effects.OneShotEffect;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.constants.*;
+import mage.game.stack.Spell;
+import mage.target.TargetSpell;
+import mage.constants.Outcome;
+import mage.game.events.GameEvent;
+import mage.players.Player;
+import mage.abilities.effects.ContinuousRuleModifyingEffectImpl;
+import mage.MageObject;
+
+/**
+ *
+ * @author skaspels
+ */
+public final class PermissionDenied extends CardImpl {
+
+    public PermissionDenied(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.INSTANT}, "{W}{U}");
+
+
+        // Counter target noncreature spell. Your opponents can't cast noncreature spells this turn.
+        this.getSpellAbility().addEffect(new PermissionDeniedCounterEffect());

```

Comment:
Use the standard `CounterTargetEffect` rather than reimplementing a custom effect.

Reply:
I have changed the code to use the standard effect.

I see that my pull request was to merge my chages into my own branch, not into the master branch. When I ask to create a pull request on magefree/mage, I don't have access to upload the file - do I "Create a New File" and copy paste my code for the card, or am I missing something on my branch that I should be doing?

I really appreciate your help in this process, there doesn't seem to be a lot of help, and although I've done coding, I've never used a branch repository in my coding (i've never programmed as part of a team for a corporate setting). You've been awesome in this process

Reply:
https://github.com/magefree/mage/pulls

If it doesn't automatically suggest to create a PR from your most recently pushed branch, then:

-> Create pull request (green button)

-> compare across forks

![image](https://github.com/skaspels/mage/assets/12538125/f153605b-0b92-4763-8361-47b6151d7bd8)